### ISSUE TITLE
Make redux thunk optional

### ISSUE THREAD

**Issue by Eptis** [toranb/ember-redux#1]
Currently redux-thunk is listed as a dependency. Shouldn't this be optional to be able to choose different middleware?

**Comment by Eptis** [comment]
By the way, nice work on the add-on, really liking it so far :+1:

**Comment by toranb** [comment]
@Eptis thanks for dropping by to give some much needed feedback about the addon! The point you bring up about middleware is totally valid and to be honest I had a struggle (internally) before I shipped the v1.0.0 that went something like this...

pros to *not* adding redux-thunk

1) if people don't want it they are not forced to use it (less moving parts/less to consider)
2) less to npm install (also w/out `redux-thunk` the npm install of this addon is less magical)
3) without it people who didn't like or want this middleware are now free to choose

cons to *not* adding redux-thunk

1) without an async middleware of some kind I personally couldn't build anything useful with redux
2) because of the point above ...not adding this means people are now forced to make a choice about "what middleware is best" (the paradox of choice argument)
3) because of both the 2 points above combined people wouldn't be productive right away with this addon

I looked at both sides of the argument and considered how projects outside of the ember ecosystem like [shasta][] chose `redux-thunk` as the "sensible default" for async middleware. Ultimately I landed on "productivity" as the big deciding factor. That said, you can still configure a middleware you want (or even add a custom stack of middleware) by adding a middleware folder (under app) in your ember-cli project. Here is one such example I did just proving it out. (sorry the docs are not ready for the "configuration" section just yet - coming soon)

https://github.com/toranb/ember-redux-example/blob/master/app/middleware/index.js

```js
import thunk from 'npm:redux-thunk';

var warnz = function thunkMiddleware({ dispatch, getState }) {
  console.warn('wait!');
  return next => action => {
    if (typeof action === 'function') {
      return action(dispatch, getState);
    }

    return next(action);
  };
};

export default [thunk, warnz];
```

I realize this wouldn't make everyone happy but I did consider the options before adding it directly.

[shasta]: https://github.com/shastajs/shasta/issues/1

**Comment by Eptis** [comment]
Those are some very valid arguments! The thing I was worried about is the increase in file size if you don't use redux-thunk, but since its 9 lines of code I think that case is pretty much neglectable.
